google-cloud-spanner/src/test/java/com/google/cloud/spanner/connection/RandomResultSetGenerator.java

```diff
@@ -40,73 +45,104 @@
+                    : Type.newBuilder().setCode(TypeCode.JSON).build(),
+                Type.newBuilder().setCode(TypeCode.BYTES).build(),
+                Type.newBuilder().setCode(TypeCode.DATE).build(),
+                Type.newBuilder().setCode(TypeCode.TIMESTAMP).build(),
+                Type.newBuilder()
+                    .setCode(TypeCode.ARRAY)
+                    .setArrayElementType(Type.newBuilder().setCode(TypeCode.BOOL))
+                    .build(),
+                Type.newBuilder()
+                    .setCode(TypeCode.ARRAY)
+                    .setArrayElementType(Type.newBuilder().setCode(TypeCode.INT64))
+                    .build(),
+                Type.newBuilder()
+                    .setCode(TypeCode.ARRAY)
+                    .setArrayElementType(Type.newBuilder().setCode(TypeCode.FLOAT64))
+                    .build(),
+                Type.newBuilder()
+                    .setCode(TypeCode.ARRAY)
+                    .setArrayElementType(
+                        dialect == Dialect.POSTGRESQL
+                            ? Type.newBuilder()
+                                .setCode(TypeCode.NUMERIC)
+                                .setTypeAnnotation(TypeAnnotationCode.PG_NUMERIC)
+                            : Type.newBuilder().setCode(TypeCode.NUMERIC))
+                    .build(),
+                Type.newBuilder()
+                    .setCode(TypeCode.ARRAY)
+                    .setArrayElementType(Type.newBuilder().setCode(TypeCode.STRING))
+                    .build(),
+                Type.newBuilder()
+                    .setCode(TypeCode.ARRAY)
+                    .setArrayElementType(
+                        dialect == Dialect.POSTGRESQL
+                            ? Type.newBuilder()
+                                .setCode(TypeCode.JSON)
+                                .setTypeAnnotation(TypeAnnotationCode.PG_JSONB)
+                            : Type.newBuilder().setCode(TypeCode.JSON))
+                    .build(),
+                Type.newBuilder()
+                    .setCode(TypeCode.ARRAY)
+                    .setArrayElementType(Type.newBuilder().setCode(TypeCode.BYTES))
+                    .build(),
+                Type.newBuilder()
+                    .setCode(TypeCode.ARRAY)
+                    .setArrayElementType(Type.newBuilder().setCode(TypeCode.DATE))
+                    .build(),
+                Type.newBuilder()
+                    .setCode(TypeCode.ARRAY)
+                    .setArrayElementType(Type.newBuilder().setCode(TypeCode.TIMESTAMP))
+                    .build()));
+
+    // TODO(harsha): add test corresponding to this in testGetAllTypesAsString()
+    // appendProtoTypes(types, dialect);
+    Type[] typeArray = new Type[types.size()];
+    typeArray = types.toArray(typeArray);
+    return typeArray;
+  }
+
+  /** To append Proto & Enum types * */
+  private static void appendProtoTypes(List<Type> types, Dialect dialect) {
```

Comment:
This does not seem to be used. Could we revert the changes in this class if they are not needed/used?

Reply:
We need this for one test. I have made the changes in this PR https://github.com/googleapis/java-spanner/pull/2845